docs/DeveloperGuide.md

```diff
@@ -4,10 +4,168 @@
+- **Storage**: The `Storage` class handles the persistence of tasks by reading from and writing to a file.
+- **Time**: The `View` class and its subclasses (`WeekView` and `MonthView`) manage the rendering and navigation of the week and month views.
+
+The following sequence diagram illustrates the high-level interaction between the components when executing a command:
+
+```puml
+@startuml
+participant Main
+participant TaskManager
+participant WeekView
+participant MonthView
+participant Storage
+
+Main -> TaskManager : executeCommand()
+activate TaskManager
+TaskManager -> WeekView : getWeekView()
+TaskManager -> MonthView : getMonthView()
+TaskManager -> Storage : loadTasksFromFile()
+TaskManager <-- Storage : tasksFromFile
+TaskManager -> WeekView : printView()
+TaskManager -> MonthView : printView()
+deactivate TaskManager
+@enduml
+```
+
+### View Switching
+
+The Calendar application supports switching between the week and month views, allowing users to visualize their tasks in different time perspectives. The switching functionality is implemented through the `WeekView` and `MonthView` classes, which extend the `View` abstract class.
+
+#### View Hierarchy
+
+The `View` abstract class provides a common interface for rendering and navigating different calendar views. It defines the following abstract methods:
+
+- `printView(TaskManager taskManager)`: This method is responsible for rendering the view to the console based on the provided `TaskManager` instance.
+- `next()`: This method advances the view to the next period (e.g., next week or next month).
+- `previous()`: This method moves the view to the previous period (e.g., previous week or previous month).
+
+The `View` class also has a protected `startOfView` field, which represents the starting date of the current view period, and a `dateFormatter` field for formatting dates.
+
+#### WeekView
+
+The `WeekView` class extends the `View` abstract class and provides an implementation for rendering and navigating the week view. It overrides the following methods from the `View` class:
+
+- `printView(TaskManager taskManager)`: This method first calculates the end of the week based on the `startOfView` date. It then prints the week header using the `UiRenderer.printWeekHeader` method and the week body using the `UiRenderer.printWeekBody` method.
+- `next()`: This method advances the `startOfView` date to the start of the next week by adding one week.
+- `previous()`: This method moves the `startOfView` date to the start of the previous week by subtracting one week.
+
+The `WeekView` class also provides the following utility methods:
+
+- `getStartOfWeek()`: Returns the `startOfView` date, which represents the start of the current week.
+- `getDateForDay(int dayOfWeek)`: Returns the date for a specific day of the week based on the `startOfView` date. The `dayOfWeek` parameter is expected to be a value between 1 (Monday) and 7 (Sunday).
```

Comment:
We may want to include code snippets and function headers for better understandability